src/realm/object-store/sync/app_service_client.hpp

```diff
@@ -36,6 +36,18 @@ class AppServiceClient {
 public:
     virtual ~AppServiceClient() = default;
 
+    /// Calls the Realm Cloud function with the provided name and arguments.
+    /// @param user The sync user to perform this request.
+    /// @param name The name of the Realm Cloud function to be called.
+    /// @param args_ejson The arguments array to be provided to the function encoded as an ejson string.
+    /// @param service_name The name of the service, this is optional.
+    /// @param completion Returns the result from the intended call, will return an Optional AppError is an
+    /// error is thrown and ejson-encoded reply if successful
+    virtual void
+    call_function(const std::shared_ptr<SyncUser>& user, const std::string& name, StringData args_ejson,
```

Comment:
Can this take args_ejson as a std::string_view rather than a StringData. StringData really should only be for strings stored in a realm.